sunpy/util/util.py

```diff
@@ -92,12 +97,13 @@ def expand_list(inp):
     ----------
     * https://stackoverflow.com/questions/2185822/expanding-elements-in-a-list/2185971#2185971
     """
-    return [item for item in expand_list_generator(inp)]
+    return list(expand_list_generator(inp))
 
 
 def expand_list_generator(inp):
+    type_union = (str | bytes | np.ndarray | dict | WCS | Header)
```

Comment:
A lot of extra are Iterable, there has to be a better class to avoid forming this list.

Reply:
Instead of checking if the input is an instance of a specific set of types, I can check if it is an instance of the `collections.abc.Collection` class, right?

Reply:
Maybe that would work?

Reply:
Are there any ways to find out if it's working or not? I'm confused

Reply:
The unit tests.